packages/react-core/src/components/Timestamp/Timestamp.tsx

```diff
@@ -77,47 +87,51 @@ export const Timestamp: React.FunctionComponent<TimestampProps> = ({
   date: dateProp = new Date(),
   dateFormat,
   datetime,
-  hasUTCTooltip,
+  displaySuffix = '',
   is12Hour,
   locale = undefined,
   timeFormat,
-  timeZoneSuffix = '',
-  tooltipProps,
-  utcSuffix = 'UTC',
+  tooltip,
   ...props
 }: TimestampProps) => {
-  const hasTooltip = hasUTCTooltip || tooltipProps?.content;
+  const hasTimeFormat = timeFormat && !customFormat;
   const formatOptions = {
     ...(dateFormat && !customFormat && { dateStyle: dateFormat }),
     ...(customFormat && { ...customFormat }),
     ...(is12Hour !== undefined && { hour12: is12Hour })
   };
-  const hasTimeFormat = timeFormat && !customFormat;
 
   const dateAsLocaleString = new Date(dateProp).toLocaleString(locale, {
     ...formatOptions,
     ...(hasTimeFormat && { timeStyle: timeFormat })
   });
+  const defaultDisplay = `${dateAsLocaleString}${displaySuffix ? ' ' + displaySuffix : ''}`;
```

Comment:
Looks like maybe this what was causing the trailing whitespace? FWIW, it's still happening with custom content when you render `children`. 

I wonder if we could do something like what you had before but use something like [.trim()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) on the `display` var

```
const display = {!children ? `${dateAsLocaleString} ${timeZoneSuffix}` : children};
```

https://github.com/patternfly/patternfly-react/blob/8c299a46c78ddc5bea2aa325c47c29477f00baba/packages/react-core/src/components/Timestamp/Timestamp.tsx#L125

Reply:
So using trim() works for the default display, but since the children prop type is ReactNode, it doesn't work as easily. Updating the children type to just a string would resolve that (and ideally only text would get passed in as custom content since that is rendered inside `time` element, with the possibility of icons in the future being rendered outside the `time` element). @tlabaj wdyt about having the children prop only allow strings?

Reply:
Ah bummer. The [<time> element accepts phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time) so IMO, it would be good to let a user put whatever they want in there, I suppose. In that case, it would be up to them to trim the whitespace if they're putting content around it where the border might extend past the timestamp text if they have a tooltip. It's an edge case anyways. I'm curious what @tlabaj thinks, too.

Reply:
We probably want to keep the type ReactNode for the reasons @mcoker mentions above. I definitely think we want the default to trim the whitespace.  Maybe it is ok for the consumer to have to format any custom display on their own... I just wonder if custom display may be a common use case.  We would probably at least want to have the example demonstrate trimming the white space.

Reply:
Michael and I had talked a bit on Slack about this, and another alternative is updating the styles in Core so that `pf-c-timestamp` has `display: inline-block` which resolved that trailing whitespace issue when I applied the style inline in my local React build.

Reply:
@mcoker @thatblindgeye  even better!